org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/EclipseSbtBuildManager.scala

```diff
@@ -77,11 +77,31 @@ class EclipseSbtBuildManager(val project: IScalaProject, settings0: Settings) ex
 
   override def build(addedOrUpdated: Set[IFile], removed: Set[IFile], pm: SubMonitor) {
     sbtReporter.reset()
+    buildInternally(addedOrUpdated, removed, pm)
+  }
+
+  override def resetReporter(): Unit = sbtReporter.reset()
+
+  override def buildBulk(addedOrUpdated: Set[IFile], removed: Set[IFile], pm: SubMonitor, bulkFilter: Set[IFile] => Set[IFile] ) {
+    buildInternally(addedOrUpdated, removed, pm, bulkFilter)
+  }
+
+  override def clean(implicit monitor: IProgressMonitor) {
+    cachePath.refreshLocal(IResource.DEPTH_ZERO, null)
+    cachePath.delete(true, false, monitor)
+    clearCached()
+  }
+
+  override def invalidateAfterLoad: Boolean = true
+
+  def findInstallation(project: IScalaProject): IScalaInstallation = project.effectiveScalaInstallation()
```

Comment:
I don't see why you need this method. I don't see any interesting difference between calling `findInstallation(project)` and `project.effectiveScalaInstallation`. Indirection makes it harder to understand the code (unless there's some abstraction/logic that you provide, but here it's not exactly the case)